src/front/glsl/error.rs

```diff
@@ -14,8 +27,8 @@ pub enum ErrorKind {
     InvalidProfile(SourceMetadata, String),
     #[error("Invalid version: {1}")]
     InvalidVersion(SourceMetadata, u64),
-    #[error("Unexpected token: {0}")]
-    InvalidToken(Token),
+    #[error("Expected {}, found '{0}'", join_with_comma(.1))]
+    InvalidToken(Token, Vec<Cow<'static, str>>),
```

Comment:
What I meant here was a `Vec<Token>`, this makes it so that if any user wants to handle the error in their code they can deal with the enum instead of a stringified representation and since in the future i intend to implement `Display` in `Token` you get the same output. And it also makes it easier to handle in the `expect` method in parser

Reply:
How would `InvalidToken(_, vec!["identifier".into()])` work in that case? Use a `TokenValue::Identifier("".to_string())` and special case identifiers to not print their inner string in the error?

Reply:
Exactly, we could maybe add a check to see if the string isn't empty but may original idea was to just print the enum kind and not it's content this would also be true for integers

Reply:
One alternative to dummy values for `Identifier`, `TypeName` etc. is what wgsl does:
Have an
```rust
enum ExpectedToken {
  Token(TokenValue),
  Identifier,
  TypeName,
  ...
}
```
and use that instead.

I implemented this in https://github.com/gfx-rs/naga/pull/934/commits/bccd6a8e0d1d26b510e2b43f002d309968012626, what do you think?

Reply:
The wgsl frontend doesn't print the enum variant names `Identifier`, `Float`, and `Paren` but `identifier`, `floating point literal` and `(` instead. Show I do that for the glsl frontend as well?

Reply:
I think the names should a little more user friendly (like the ones in the wgsl frontend as you mentioned) and instead of `expected_str` you should implement `Display`. For tokens if you want to add pretty printing please do it in a separate PR since it's quite a big change

Reply:
[Done](https://github.com/gfx-rs/naga/pull/934/commits/1dee90ece62ceb6bd5c6bb2d0c623c306f87a238)